### PR TITLE
Souce Refactor. Abstracted redundant functionality inside two Create Arr...

### PR DESCRIPTION
**Pull request by krishnanm86**
...ay methods in cjson.cc

### CODE REVIEW

File: src/http/json/cJSON.cc [hunk 1]
```diff
@@ -731,21 +732,13 @@ cJSON *cJSON_CreateIntArray(int *numbers,int count) {
     return a;
 }
 
+cJSON *cJSON_CreateIntArray(int *numbers,int count) {
+    return cJSON_CreateNumArray(numbers,count);
+}
+
 cJSON *cJSON_CreateDoubleArray(double *numbers,int count) {
     int i;
```
Comment (danielmewes) [inline src/http/json/cJSON.cc]: Can you delete this `int i;` line as well please?

**Comment by danielmewes** [comment]
Thanks for the pull request @krishnanm86 !
I added one minor comment. Otherwise this looks good.

File: src/http/json/cJSON.cc (hunk 1 above)
Comment (krishnanm86) [inline src/http/json/cJSON.cc]: Done. Hopefully it can be merged now.

**Comment by danielmewes** [comment]
Ah I meant the other `int i;`, the one in `cJSON_CreateDoubleArray()` https://github.com/rethinkdb/rethinkdb/pull/3820/files#diff-f426bd3996a091dd640057ede5a73c22R739

**Comment by krishnanm86** [comment]
Removed the unneeded int and merged one more redundant pair. Hope this saves some code space and improves readability. the idea is hopefully that bugs if introduced need to be fixed at one location only.

**Comment by danielmewes** [comment]
Thanks. I'd actually prefer to not have that last change in `varint.cc`.

The reason is that this code is part of Google's S2 library (we just copied it into our code base to make compilation easier and integrate it better). Having changes in that code will make it more difficult for us to merge upstream changes back into our code base.
You can try to submit that change to https://code.google.com/p/s2-geometry-library/

(with cJSON it's actually similar but we have already adapted it somewhat, so there it's less critical)

If you revert the last commit bac742b this will be good to merge.

**Comment by krishnanm86** [comment]
Done

**Comment by danielmewes** [comment]
Perfect, thanks again.

**Pull request merged**